tests/core/test_lightning_optimizer.py

```diff
@@ -206,6 +207,55 @@ def test_state(tmpdir):
     assert optimizer.state == lightning_optimizer.state
 
 
+def test_lightning_optimizer_with_wrong_optimizer_interface(tmpdir):
+    class OptimizerWrapper(object):
```

Comment:
LGTM. Inheriting just `optim.Adam` and overriding the `step` method would have worked too. Also can you add an assertion to check that closure is called, maybe by overriding backward method in `TestLightningOptimizerModel`.

Reply:
Yes, I agree.